server/app/auth/oidc/IdTokens.java

```diff
@@ -0,0 +1,68 @@
+package auth.oidc;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.nimbusds.jwt.JWT;
+import com.nimbusds.jwt.JWTClaimsSet;
+import com.nimbusds.jwt.JWTParser;
+import java.text.ParseException;
+import java.time.Clock;
+import java.util.Date;
+import java.util.Optional;
+
+/** Class that manages stored ID tokens for an Account. */
+public final class IdTokens {
+  private SerializedIdTokens serializedIdTokens;
+
+  private Clock clock;
+
+  public IdTokens(Clock clock, SerializedIdTokens serializedIdTokens) {
+    this.clock = checkNotNull(clock);
+    this.serializedIdTokens = checkNotNull(serializedIdTokens);
+  }
+
+  public Optional<String> getIdToken(String sessionId) {
+    if (!serializedIdTokens.containsKey(sessionId)) {
+      return Optional.empty();
+    }
+    return Optional.of(serializedIdTokens.get(sessionId));
```

Comment:
**nitpick (non-blocking):** could maybe do this instead?
```suggestion
    return Optional.ofNullable(serializedIdTokens.getOrDefault(sessionId, null));
```

Reply:
Nice, thanks!